### PR TITLE
cases by list and not a free for all

### PR DESCRIPTION
**Pull request by NirWilner**
A script will be run on the server that will transfer the current format to the new one

### CODE REVIEW

File: app/Utils/DBHandler.py [hunk 1]
```diff
@@ -37,6 +37,13 @@ def add_item(self, platform: int, item_type: int, username: str, date: datetime,
             link (str): A link that leads to that item
             image (str): ""
         """
+        for case in cases:
+            if not self.cases_handler.is_case_exists(case):
+                #self.cases_handler.add_case(case)
+                raise Exception("This case doesnt exist only an admin can add a case")
+        
+        cases = [self.cases_handler.get_case_by_case_name(i) for i in cases]
```
Comment (sapirrap) [inline app/Utils/DBHandler.py:45]: Eitan from algoritmics class will be ssad

File: app/__init__.py [hunk 2]
```diff
@@ -44,6 +44,6 @@
 api.add_resource(AdminResource, "/admin")
 api.add_resource(TasksResource, "/tasks")
 api.add_resource(ChecklistsResource, "/checklists")
-api.add_resource(CasesResource, "/cases")
+api.add_resource(CasesResource, "/cases", "/cases/<param>")
```
Comment (sapirrap) [inline app/__init__.py:47]: should be in admin probably

File: app/__init__.py [hunk 3]
```diff
@@ -44,6 +44,6 @@
 api.add_resource(AdminResource, "/admin")
```
Comment (sapirrap) [inline app/__init__.py:44]: admin should be settings probably

File: app/Cases/Resource/CasesResource.py [hunk 4]
```diff
@@ -16,10 +17,16 @@
...
+    def get(self, param=""):
         if not current_user.is_authenticated:
             return make_response(jsonify(WebAnalysisResponse.NOT_LOGGED_IN.value), HTTPStatus.OK.value)
         
+        #if the user requests a list of all the posibble cases, return them sorted by the most popular case
+        if param == "all":
+            cases_and_count_dict = current_user.db_handler.graph_handler.get_all_unique_cases_and_count()
+            sorted_cases = [i["case"] for i in sorted(cases_and_count_dict, reverse=True, key=lambda d: d["count"])]
+            return make_response(jsonify(sorted_cases), HTTPStatus.OK.value)
+
```
[the remark this answers is not in the document]
Comment (NirWilner) [reply app/Cases/Resource/CasesResource.py:29]: Bad

File: app/Cases/Handles/CasesHandler.py [hunk 5]
```diff
@@ -83,6 +83,14 @@ def get_case_by_case_name(self, case_name: str):
         """
         return self.db.cases.find_one({CaseFields.CONTENT_FIELD.value: case_name})
     
+    def get_all_cases(self) -> list:
+        """A function that gives all the possible cases
+
+        Returns:
+            list: _description_
```
Comment (sapirrap) [inline app/Cases/Handles/CasesHandler.py]: fix

**Review by sapirrap — CHANGES_REQUESTED** [review]

**Review by sapirrap — APPROVED** [review]

**Pull request merged**